src/components/course/data/utils.jsx

```diff
@@ -407,10 +409,344 @@ export function shortenString(str, maxLength, suffix, separator = ' ') {
+    return undefined;
+  }
+
+  const hasExpiredSubscriptions = subscriptionsApplicableToCourse.every(
+    subscription => subscription.daysUntilExpirationIncludingRenewals < 0,
+  );
+  const hasExhaustedSubscriptions = subscriptionsApplicableToCourse.every(
+    subscription => subscription?.licenses?.unassigned === 0,
+  );
+  const applicableSubscriptionNonExpiredNonExhausted = subscriptionsApplicableToCourse.find(
+    subscription => subscription.daysUntilExpirationIncludingRenewals >= 0 && subscription?.licenses?.unassigned > 0,
+  );
+
+  if (hasExpiredSubscriptions) {
+    // If customer's subscription plan(s) containing the course being viewed have expired,
+    // change `reasonType` to use the `SUBSCRIPTION_EXPIRED` message.
+    return parseReasonTypeBasedOnEnterpriseAdmins({
+      hasEnterpriseAdminUsers,
+      reasonTypes: {
+        hasAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_EXPIRED,
+        hasNoAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_EXPIRED_NO_ADMINS,
+      },
+    });
+  }
+
+  if (hasExhaustedSubscriptions) {
+    // If customer's subscription plan(s) containing the course being viewed no longer have
+    // any remaining seats, change `reasonType` to use the `SUBSCRIPTION_SEATS_EXHAUSTED` message.
+    return parseReasonTypeBasedOnEnterpriseAdmins({
+      hasEnterpriseAdminUsers,
+      reasonTypes: {
+        hasAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_SEATS_EXHAUSTED,
+        hasNoAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_SEATS_EXHAUSTED_NO_ADMINS,
+      },
+    });
+  }
+
+  if (subscriptionLicense?.status === LICENSE_STATUS.REVOKED) {
+    // If learner's subscription license is revoked, change `reasonType` to use
+    // the `SUBSCRIPTION_DEACTIVATED` message.
+    return DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_DEACTIVATED;
+  }
+
+  if (applicableSubscriptionNonExpiredNonExhausted) {
+    // If customer has a subscription plan(s) containing the course being viewed that is not expired
+    // nor exhausted, change `reasonType` to use the `SUBSCRIPTION_LICENSE_NOT_ASSIGNED` message.
+    return parseReasonTypeBasedOnEnterpriseAdmins({
+      hasEnterpriseAdminUsers,
+      reasonTypes: {
+        hasAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_LICENSE_NOT_ASSIGNED,
+        hasNoAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_LICENSE_NOT_ASSIGNED_NO_ADMINS,
+      },
+    });
+  }
+
+  // There is no applicable subscriptions-related reason for disabled enrollment.
+  return undefined;
+};
+
+export const getEnterpriseOffersDisabledEnrollmentReasonType = ({
+  enterpriseOffers,
+  catalogsWithCourse,
+}) => {
+  if (!enterpriseOffers || enterpriseOffers.length === 0) {
+    return DISABLED_ENROLL_REASON_TYPES.NO_ENTERPRISE_OFFERS;
+  }
+
+  const offersForCourse = enterpriseOffers.filter(offer => catalogsWithCourse.includes(offer.enterpriseCatalogUuid));
+  if (!offersForCourse.length) {
+    return DISABLED_ENROLL_REASON_TYPES.COURSE_NOT_IN_ENTERPRISE_OFFERS;
```

Comment:
Note: There is no `COURSE_NOT_IN_ENTERPRISE_OFFERS` under `DISABLED_ENROLL_REASON_TYPES`.

Related, I don't believe this function necessarily needs to return a `reasonType` around having no offers applicable to the catalogs containing the course?

Rationale: If there are no offers applicable to any of the `catalogsWithCourse` and the user has no other non-offer subsidy applicable to the course, the default `reasonType` set by the `getMissingApplicableSubsidyReason` function is either `DISABLED_ENROLL_REASON_TYPES.NO_SUBSIDY_NO_ADMINS` or `DISABLED_ENROLL_REASON_TYPES.NO_SUBSIDY` depending on whether the enterprise customer has administrators.

We don't want to expose a message to the user saying  "no offers with course in associated catalog.". See https://www.figma.com/file/o4qkdga0IRIqpdF73JGNCA/Course-about-page-fund-error-state?type=design&node-id=614-3906&mode=dev for the various intended error messages to show across subsidy types.